src/base/static/libs/maps/mapboxgl-provider.js

```diff
@@ -4,8 +4,131 @@ import "mapbox-gl/dist/mapbox-gl.css";
 
 import VectorTileClient from "../../client/vector-tile-client";
 
+import constants from "../../constants";
+
 mapboxgl.accessToken = MAP_PROVIDER_TOKEN;
 
+const appendFilters = (existingFilters, ...filtersToAdd) => {
+  const newFilters = filtersToAdd.reduce(
+    (newFilters, filterToAdd) => [...newFilters, filterToAdd],
+    [existingFilters],
+  );
+
+  // If an existing filter does not already start with the logical AND
+  // operator "all", we need to prepend it before we add a new filter.
+  if (newFilters[0] !== "all") {
+    newFilters.unshift("all");
+  }
+
+  return newFilters;
+};
+
+const layerToSymbol = layerConfigs => {
+  return layerConfigs.map(layerConfig => {
```

Comment:
Technically these functions are mapping layer**s** to a symbol. I think it would be more clear to keep these functions as simple as possible by mapping a single `layerConfig` to their mapbox representation. Then the caller can call `layers.map(layerToSymbol)`.